### PR TITLE
dotCMS/core#12383 Create a WYSIWYG/CODE/TEXT component

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: .angular-cli.json [hunk 1]
```diff
@@ -16,6 +16,11 @@
             "testTsconfig": "tsconfig.spec.json",
             "prefix": "app",
             "styles": ["styles.scss"],
+            "stylePreprocessorOptions": {
```
Comment (fmontes) [inline .angular-cli.json]: This it's what allows me to do this: https://github.com/dotCMS/core-web/pull/419/files#diff-a09f6a8154f077be31dd14760c18666aR1

File: src/app/modules.ts [hunk 2]
```diff
@@ -25,13 +25,15 @@ import {
     ToolbarModule,
     TreeTableModule
 } from 'primeng/primeng';
+import { DotTextareaContentModule } from './view/components/_common/dot-textarea-content/dot-textarea-content.module';
```
Comment (fmontes) [inline src/app/modules.ts]: Just for the pattern library, after @hmoreras changes we can move it to the pattern library module.

File: .angular-cli.json (hunk 1 above)
Comment (freddyucv) [reply .angular-cli.json]: cool

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/values-property/values-property.component.spec.ts [hunk 3]
```diff
@@ -33,6 +34,7 @@ describe('ValuesPropertyComponent', () => {
                 ValuesPropertyComponent
             ],
             imports: [
+                DotTextareaContentModule
```
Comment (freddyucv) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/values-property/values-property.component.spec.ts]: you have to test some more here, for example if the dot-textarea-content exist and has the right attributes

File: src/app/view/components/_common/dot-textarea-content/dot-textarea-content.component.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    propagateChange = (_: any) => {};
+
+    constructor() {
+    }
+
+    ngOnInit() {
+        if (this.show.length) {
+            this.show.forEach(item => {
+                this.selectOptions.push({
```
Comment (freddyucv) [inline src/app/view/components/_common/dot-textarea-content/dot-textarea-content.component.ts]: maybe we should validate that item is a valid options, we could compare with the default selectOptions array.

File: src/app/view/components/_common/dot-textarea-content/dot-textarea-content.component.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                    label: `${item.charAt(0).toUpperCase()}${item.slice(1)}`,
+                    value: item
+                });
+            });
+        } else {
+            this.selectOptions = [
+                { label: 'Plain', value: 'plain' },
+                { label: 'Code', value: 'code' },
+                { label: 'WYSIWYG', value: 'wysiwyg' }
+            ];
```
Comment (freddyucv) [inline src/app/view/components/_common/dot-textarea-content/dot-textarea-content.component.ts]: the default value should be a static class attribute

File: src/app/view/components/_common/dot-textarea-content/dot-textarea-content.component.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    /**
+     * Update model with external value
+     *
+     * @param {string} value
+     * @memberof DotTextareaContentComponent
+     */
+    writeValue(value: string): void {
+        if (value) {
```
Comment (freddyucv) [inline src/app/view/components/_common/dot-textarea-content/dot-textarea-content.component.ts]: what happen if I clean up the component's value?

Comment (fmontes) [reply src/app/view/components/_common/dot-textarea-content/dot-textarea-content.component.ts]: Work as expected, value it's empty string.

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/values-property/values-property.component.spec.ts (hunk 3 above)
Comment (fmontes) [reply src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/values-property/values-property.component.spec.ts]: Done

**Pull request merged**